shoop/core/order_creator/_creator.py

```diff
@@ -248,3 +234,28 @@ def process_order_before_lines(self, source, order):
     def process_order_after_lines(self, source, order):
         # Subclass hook
         pass
+
+
+class OrderCreator(OrderProcessor):
+
+    def __init__(self, request=None):
+        """
+        Initialize order creator.
+
+        :type request: django.http.HttpRequest|None
+        :param request:
+          Optional request object for backward compatibility.  Passing
+          non-None value is DEPRECATED.
+        """
+        if request is not None:
+            warnings.warn(
+                "Initializing OrderCreator with a request is deprecated",
+                DeprecationWarning, stacklevel=2)
+
+    def create_order(self, order_source):
+        data = self.get_source_base_data(order_source)
+        order = Order(**data)
+        order.save()
+        order = self.finalize_creation(order, order_source)
+        order_creator_finished.send(sender=type(self), order=order, source=order_source)
```

Comment:
maybe the original idea was to have it before processing lines because the "base order is finished" in that step. If @suutari-ai doesn't find anything to complain about this, I'm good with it.